c7n/resources/rds.py

```diff
@@ -1819,3 +1820,56 @@ class resource_type(TypeInfo):
         universal_taggable = object()
 
     augment = universal_augment
+
+
+@filters.register('consecutive-snapshots')
+class ConsecutiveSnapshots(Filter):
+    """Returns instances where number of consective daily snapshots is equal to/or greater than n days.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: rds-daily-snapshot-count
+                resource: rds
+                filters:
+                  - type: consecutive-snapshots
+                    days: 7
+                    exclude_engines: ["postgres"]
+    """
+    schema = type_schema('consecutive-snapshots', days={'type': 'number', 'minimum': 1},
+     exclude_engines={'type': 'array', 'items': {'type': 'string'}}, required=['days'])
+    permissions = ('rds:DescribeDBSnapshots', 'rds:DescribeDBInstances')
+
+    def enhance_resources(self, resources):
```

Comment:
For the sake of consistency, it would be helpful to call this method `augment` and have it modify resources in place rather than returning a new object. Putting this data into a key with an `c7n:` prefix (such as `c7n:DBSnapshots`) helps it stand out from keys that are part of the main API response. Rationale:

* Snapshot data will be part of policy execution output (`resources.json`) and accessible to subsequent filters in the policy
* We can avoid fetching snapshots more than once (admittedly that seems unlikely here)

There some good examples of filter-level augments, such as the account-level [s3-public-block](https://github.com/cloud-custodian/cloud-custodian/blob/002fa72de3cb3faf6309b0608a4f4216c2f1d571/c7n/resources/account.py#L1498-L1522) filter.

Reply:
great suggestion  and thanks for the guidance and explanation. Hopefully this fix is in line with your recommendation.